### PR TITLE
Feature/filter on played

### LINKED ISSUE #6: As a gamer, I wat to be able to track which games I've previously played
* Add a checkbox on the game object for the user to use OR count only reviewed games as played
* Save which games each user has played in the backend (make sure to follow GDPR)
* Add a way to filter to only games previously played
* * Add a way to filter to only games not played previously

### PR DESCRIPTION
**Pull request by x183**
Added ability to filter games on whether they were played by the user or not, fixes #6

### CODE REVIEW

**Comment by Supergamer1337** [comment]
Why does it have a lot of rating stuff?

File: backend/src/routers/gameRouter.ts [hunk 1]
```diff
@@ -280,6 +283,12 @@ gameRouter.post(
 				}).length > 0
 			}
 		}
+		if (body.isPlayed != undefined) {
+			res.status(200).json(formattedGames.filter((game) => {
+				return game.isPlayed == body.isPlayed;
+			}));
+			return;
+		}
```
Comment (Supergamer1337) [inline backend/src/routers/gameRouter.ts:291]: It's not possible to do this with a DB call?

File: frontend/src/components/GamesList/GamesList.tsx [hunk 2]
```diff
@@ -60,7 +62,9 @@ const GamesList: FC<GamesListProps> = () => {
 		if (playtimeMin) searchFilter.playtimeMin = playtimeMin;
 		if (playerCount) searchFilter.playerCount = playerCount;
 		if (owner) searchFilter.owner = owner;
+		if (isPlayed != undefined) searchFilter.isPlayed = isPlayed;
 		postData(searchFilter);
+		console.log(searchFilter);
```
Comment (Supergamer1337) [inline frontend/src/components/GamesList/GamesList.tsx:67]: Why is this still here?
```suggestion
```

Comment (Supergamer1337) [inline frontend/src/components/GamesList/GamesList.tsx:67]: Why is this still here?
```suggestion
```

**Review by Supergamer1337 — CHANGES_REQUESTED** [review]

**Comment by x183** [comment]
wrong pr, sorry 😓

**Comment by x183** [comment]
I'll continue with this after the current sprint is finished

**Pull request closed without merging**